adapters/repos/db/index.go

```diff
@@ -269,21 +269,21 @@ func NewIndex(ctx context.Context, cfg IndexConfig,
 		}
 
 		shardName := shardName // prevent loop variable capture
-		eg.Go(func() error {
-			shard, err := NewLazyLoadShard(ctx, promMetrics, shardName, index, class, jobQueueCh, indexCheckpoints)
-			if err != nil {
-				return fmt.Errorf("init shard %s of index %s: %w", shardName, index.ID(), err)
-			}
 
-			index.shards.Store(shardName, shard)
-			return nil
-		})
-	}
+		shard, err := NewLazyLoadShard(ctx, promMetrics, shardName, index, class, jobQueueCh, indexCheckpoints)
+		if err != nil {
+			return nil, fmt.Errorf("init shard %s of index %s: %w", shardName, index.ID(), err)
+		}
 
-	if err := eg.Wait(); err != nil {
-		return nil, err
+		index.shards.Store(shardName, shard)
 	}
 
+	go index.ForEachShard(func(name string, shard ShardLike) error {
+		shard.Load(context.Background())
+		time.Sleep(1 * time.Second)
```

Comment:
Should make this configurable, maybe?

Reply:
Class config is probably cluttering things for the average user, I guess we can read another environment variable